WordPress/Classes/ViewRelated/Menus/MenuDetailsViewController.m

```diff
@@ -0,0 +1,232 @@
+            self.textField.enabled = NO;
+            self.textFieldDesignIcon.hidden = YES;
+            self.trashButton.hidden = YES;
+        }
+    }
+}
+
+- (void)hideTextFieldKeyboard
+{
+    [self.textField resignFirstResponder];
+}
+
+- (void)updateTextFieldDesignIconPositioning
+{
+    CGSize textSize = [self.textField.text sizeWithAttributes:@{NSFontAttributeName: self.textField.font}];
+    CGRect editingRect = [self.textField textRectForBounds:self.textField.bounds];
+    CGFloat leadingConstant = editingRect.origin.x + textSize.width;
+    if (leadingConstant > self.textField.frame.size.width) {
+        leadingConstant = self.textField.frame.size.width;
+        leadingConstant += 1.0; // padding
+    } else  {
+        leadingConstant += 6.0; // padding
+    }
+    
+    self.textFieldDesignIconLeadingConstraint.constant = ceilf(leadingConstant);
+    [self.textFieldDesignIcon setNeedsLayout];
+}
+
+- (void)showTextFieldEditingState:(NSTimeInterval)duration
+{
+    [UIView animateWithDuration:duration animations:^{
+        
+        self.doneButton.alpha = 1.0;
+        self.textFieldDesignIcon.hidden = YES;
+        self.textFieldDesignView.backgroundColor = [UIColor colorWithWhite:1.0 alpha:0.6];
+        
+    } completion:nil];
+}
+
+- (void)hideTextFieldEditingState:(NSTimeInterval)duration
+{
+    [UIView animateWithDuration:duration animations:^{
+        
+        self.textFieldDesignView.backgroundColor = [UIColor clearColor];
+        self.doneButton.alpha = 0.0;
+
+    } completion:nil];
+}
+
+- (void)traitCollectionDidChange:(UITraitCollection *)previousTraitCollection
+{
+    [super traitCollectionDidChange:previousTraitCollection];
+    
+    [self updateTextFieldDesignIconPositioning];
+}
+
+- (BOOL)resignFirstResponder
+{
+    return [self.textField resignFirstResponder];
+}
+
+#pragma mark - buttons
+
+- (void)doneButtonPressed
+{
+    [self.textField resignFirstResponder];
+}
+
+- (void)trashButtonPressed
+{
+    [self.delegate detailsViewControllerSelectedToDeleteMenu:self];
+}
+
+#pragma mark - delegate helpers
+
+- (void)tellDelegateMenuNameChanged
+{
+    [self.delegate detailsViewControllerUpdatedMenuName:self];
+}
+
+#pragma mark - UITextField
+
+- (void)textFieldValueChanged:(UITextField *)textField
+{
+    if (textField.text.length) {
+        // Update the Menu name.
+        self.menu.name = textField.text;
+    } else {
+        // Restore the original menu name.
+        self.menu.name = self.editingBeginningName;
+        // Show it as a placeholder for now.
+        textField.placeholder = self.menu.name;
+    }
+    [self tellDelegateMenuNameChanged];
+    if (!self.hasTrackedEditingChanges) {
+        [WPAppAnalytics track:WPAnalyticsStatMenusUpdatedMenuName withBlog:self.menu.blog];
+        self.hasTrackedEditingChanges = YES;
+    }
+}
+
+- (void)textFieldDidBeginEditing:(UITextField *)textField
+{
+    self.editingBeginningName = textField.text;
+    [self showTextFieldEditingState:0.3];
```

Comment:
It looks like `showTextFieldEditingState` and `hideTextFieldEditingState` are each only called once.  It might make more sense to just call the method without passing the parameter, and instead define the `0.3` as a const that is referenced inside each method implementaiton.